src/main/scala/chisel3/util/experimental/decode/TruthTable.scala

```diff
@@ -31,10 +31,14 @@ object TruthTable {
 
   /** Convert a table and default output into a [[TruthTable]]. */
   def apply(table: Iterable[(BitPat, BitPat)], default: BitPat, sort: Boolean = true): TruthTable = {
-    require(table.map(_._1.getWidth).toSet.size == 1, "input width not equal.")
+    val inputWidth = table.map(_._1.getWidth).max
     require(table.map(_._2.getWidth).toSet.size == 1, "output width not equal.")
     val outputWidth = table.map(_._2.getWidth).head
-    val mergedTable = table
+    val mergedTable = table.map {
+      case (in, out) =>
+        // pad input signals.
+        (BitPat.dontCare(inputWidth - in.getWidth) ## in, out)
```

Comment:
Also is don't care padding correct? If someone creates a `BitPat` from `0.U`, the implication to me would be zero padding, not dontcare padding.

Reply:
My bad... I'll change this after wake up.

Reply:
Sorry, it seems I took 6 days to weak up :(